kernel/src/hil/qdec.rs

```diff
@@ -0,0 +1,33 @@
+//! Interface for a Qdec compatible chip
+//!
+//! This trait provides a stanfard interface for chips with a
+//! quadrature decoder. Note this interface is experimental and
+//! may need further updates once implemented on additional chips
+
+use crate::returncode::ReturnCode;
+
+pub trait QdecDriver {
+    /// Sets the client which will receive interrupts
+    fn set_client(&self, client: &'static dyn QdecClient);
+
+    /// Enables the SAMPLERDY interrupt
+    fn enable_interrupts(&self) -> ReturnCode;
+
+    /// Enables the Qdec, returning error if Qdec does not exist
+    fn enable_qdec(&self) -> ReturnCode;
+
+    /// Checks if the qdec has been enabled
+    fn enabled(&self) -> ReturnCode;
```

Comment:
Should there be a corresponding disable method?

Reply:
Thanks for pointing this out! I will make sure to add a disable function here shortly.

Reply:
I'm not sure this is documented anywhere except prior HIL discussion threads, but I think we have decided that an `is_enabled` check is a bit an anti-pattern for devices.

Callers should know whether the device is enabled or not (i.e. if you intend to use it, you need to enable it first) and if there are multiple users, then there should be a virtualization layer that multiplexes hardware access and handles {en/dis}able correctly. We have the `EOFF` return code for the case where there is a runtime bug as well.

In summary: I think we should remove the `enabled()` method, and add a `disable` method.

Reply:
Good point!`enabled()` has been removed from the HIL